src/rtl_entropy.c

```diff
@@ -84,90 +88,233 @@ unsigned int buffercounter = 0;
 AES_KEY wctx;
 EVP_CIPHER_CTX en;
 
+int read_config_file (FILE * infile, char ***config_options);
+void * Alloc (size_t len);
+char * StrnDup (char *str);
+char * StrMem (size_t slen);
 
-void usage(void) {
-  fprintf(stderr,
-	  "rtl_entropy, a high quality entropy source using RTL2832 based DVB-T receivers\n\n"
-	  "Usage: rtl_entropy [options]\n"
-	  "\t-a Set gain (default: max for dongle)\n"
-	  "\t-d Device index (default: 0)\n"
-	  "\t-e Encrypt output\n"
-	  "\t-f Set frequency to listen (default: 70MHz )\n"
-	  "\t-s Samplerate (default: 3200000 Hz)\n");
-  fprintf(stderr,
-	  "\t-o Output file (default: STDOUT, /var/run/rtl_entropy.fifo for daemon mode (-b))\n"
+void usage(void)
+{ fprintf(stderr, "rtl_entropy, a high quality entropy source using RTL2832 based DVB-T receivers\n\n");
+  fprintf(stderr, "Usage: rtl_entropy [options]\n");
+  fprintf(stderr, "--amplify,       -a []  Set gain (default: max for dongle)\n");
 #ifndef __APPLE__
-	  "\t-p PID file (default: /var/run/rtl_entropy.pid)\n"
-	  "\t-b Daemonize\n"
-	  "\t-u User to run as (default: rtl_entropy)\n"
-	  "\t-g Group to run as (default: rtl_entropy)\n"
+  fprintf(stderr, "--daemonize,     -b     Daemonize\n");
 #endif
-	  );
+  fprintf(stderr, "--config_file,   -c []  Configuration file (defaults: /etc/rtl_entropy.conf, /etc/sysconfig/rtl_entropy.conf)\n");
+  fprintf(stderr, "--device_idx,    -d []  Device index (default: %i)\n", dev_index);
+  fprintf(stderr, "--encrpyt,       -e     Encrypt output\n");
+  fprintf(stderr, "--frequency,     -f []  Set frequency to listen (default: %i MHz)\n", frequency);
+#ifndef __APPLE__
+  fprintf(stderr, "--group,         -g []  Group to run as (default: rtl_entropy)\n");
+#endif
+  fprintf(stderr, "--help,          -h     This help. (Default no)\n");
+  fprintf(stderr, "--output_file,   -o []  Output file (default: STDOUT, /var/run/rtl_entropy.fifo for daemon mode (-b))\n");
+#ifndef __APPLE__
+  fprintf(stderr, "--pid_file,      -p []  PID file (default: /var/run/rtl_entropy.pid)\n");
+#endif
+  fprintf(stderr, "--quiet,         -q []  quiet level, how much output to print, 0-3 (default: %i, print all)\n", gflags_quiet);
+  fprintf(stderr, "--sample_rate,   -s []  Samplerate (default: %i Hz)\n", samp_rate);
+#ifndef __APPLE__
+  fprintf(stderr, "--user,          -u []  User to run as (default: rtl_entropy)\n");
+#endif
+  fprintf(stderr, "Configuration file at /etc/sysconfig/rtl_entropy has more detail and sample values.\n");
```

Comment:
If I understand what you're asking, I think it doesn't matter.  This lets distros that don't have sysconfig put the config in /etc directly.  But if a distro wants to hide it in /etc/sysconfig it will still be found.  When the program doesn't find one in /etc it will look next in /etc/sysconfig.  But as soon as it finds one, it stops looking.  The hierarchy is command line, /etc, /etc/sysconfig.